plugins/dialog/plugin.js

```diff
@@ -656,11 +664,8 @@ CKEDITOR.DIALOG_STATE_BUSY = 2;
 				var id = target.$.id;
 				this.selectPage( id.substring( 4, id.lastIndexOf( '_' ) ) );
 
-				if ( this._.tabBarMode ) {
-					this._.tabBarMode = false;
-					this._.currentFocusIndex = -1;
-					changeFocus( 1 );
-				}
+				focusActiveTab( this );
```

Comment:
> If you focus a tab by keyboard (e.g. `Tab` / `Shift + Tab`) you are in the `tabBarMode` (keyboard navigation allowed), if you then mouse-click on any tab it will move focus to next focusable element (so first dialog input in active tab).

Could you report it (or see if we have already reported 🤔). Then we can also document and add to changelog closing this issue.

Reply:
I extracted the issue related to mouse focus here #3689